docs/DeveloperGuide.md

```diff
@@ -391,6 +391,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+   Use case ends.
+
+**Extensions**
+* 2a. User enters invalid information (e.g., telegram name longer than 32 characters).
+    * 2a1. PlanPal displays an error message.
+    * 2a2. User re-enters valid information.
+
+      Use case resumes from step 3.
+
+**UC09: Remove a contact from an event**
+
+**MSS**
+1. User specifies the event they want to modify.
+2. User specifies the contact they wish to remove from the event.
+3. PlanPal removes the contact from the event and displays a success message.
+
+   Use case ends.
+
+**Extensions**
+* 2a. The specified contact is not part of the event.
+    * 2a1. PlanPal displays an error message indicating the contact was not found in the event.
+
+      Use case ends.
+
+**UC10: Clear all events**
+
+**MSS**
+1. User specifies the command to clear all events.
+2. PlanPal deletes all events and displays a success message.
+
+   Use case ends.
+
+**Extensions**
+* 1a. There are no events to clear.
+    * 1a1. PlanPal displays a message indicating there are no events.
+
+      Use case ends.
+
+**UC11: Enter Search Mode**
+
+**MSS**
+1. User specifies the command to enter search mode.
+2. PlanPal activates search mode.
+
+   Use case ends.
+
+**Extensions**
+* 1a. User is already in search mode.
+    * 1a1. PlanPal displays a message indicating the user is already in search mode.
+
+      Use case ends.
+
+**UC12: Perform a search in Search Mode**
+
+**Preconditions**: User is in search mode.
+
+**MSS**
+1. User specifies search criteria (e.g., name, role, email).
+2. PlanPal filters the list of contacts based on the criteria and displays the results.
+
+   Use case ends.
+
+**Extensions**
+* 1a. User provides invalid or incomplete search criteria.
+    * 1a1. PlanPal displays an error message and prompts the user to provide valid criteria.
+    * 1a2. User re-enters valid search criteria.
+
+      Use case resumes from step 2.
+
+* 2a. No contacts match the search criteria.
+    * 2a1. PlanPal displays a message indicating no matches were found.
+
+      Use case ends.
+
+**UC13: Add a contact to the excluded list**
+
+**Preconditions**: User is in search mode.
+
+**MSS**
+1. User specifies the command to exclude a contact.
+2. User selects a contact from the current search results.
+3. PlanPal adds the contact to the excluded list and removes it from the search results, displaying a success message.
+
+   Use case ends.
+
+**Extensions**
+* 2a. User specifies an invalid or non-existent contact.
+    * 2a1. PlanPal displays an error message indicating the contact is invalid.
+    * 2a2. User re-enters a valid contact.
+
+      Use case resumes from step 3.
+
+**UC14: Manage excluded contacts in search mode**
+
+**Preconditions**: User is in search mode.
+
+**MSS**
+1. User enters search mode.
+2. User specifies a contact to exclude from search results.
+3. PlanPal excludes the contact from future searches and displays a success message.
+
+   Use case ends.
+
+**Extensions**
```

Comment:
Extension 2a seems to follow path of UC13, maybe could remove to focus UC14 on the clearing of exclusion or inclusion path to UC 13 instead